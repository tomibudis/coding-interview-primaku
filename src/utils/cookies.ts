import Cookies from "js-cookie";

const COOKIE_TOKEN_NAME = "jitera-token";

export const setCookieToken = (token: string) => {
  return Cookies.set(COOKIE_TOKEN_NAME, token);
};

export const getCookieToken = () => {
  return Cookies.get(COOKIE_TOKEN_NAME);
};

export const clearCookieToken = () => {
  return Cookies.remove(COOKIE_TOKEN_NAME);
};
