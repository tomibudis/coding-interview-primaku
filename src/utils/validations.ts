import * as yup from "yup";

export const exampleValidation = yup
  .object({
    email: yup.string().email().required(),
    password: yup.string().min(3).required(),
  })
  .required();
