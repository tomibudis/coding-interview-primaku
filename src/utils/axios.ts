import axios, { AxiosRequestConfig } from "axios";

import { getCookieToken } from "./cookies";

const customAxios = axios.create({
  baseURL: process.env.NEXT_PUBLIC_API_URL,
});

customAxios.interceptors.request.use((config: AxiosRequestConfig) => {
  const token = getCookieToken();

  return {
    ...config,
    headers: {
      ...config.headers,
      Authorization: `Bearer ${token}`,
    },
  };
});

export default customAxios;
