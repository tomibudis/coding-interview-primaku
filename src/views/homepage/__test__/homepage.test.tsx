import React from "react";

import { render } from "~/utils/test-utils";

import Homepage from "../homepage";

describe("homepage render", () => {
  it("match with snapshot", () => {
    const { baseElement } = render(<Homepage />);
    expect(baseElement).toMatchSnapshot();
  });
});
