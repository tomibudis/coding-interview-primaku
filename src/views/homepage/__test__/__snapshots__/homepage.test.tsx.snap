// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage render match with snapshot 1`] = `
<body>
  <div>
    <div
      class="w-screen h-screen flex flex-col justify-center items-center"
    >
      <h1>
        Counter App
      </h1>
      <div
        class="flex items-center mt-4"
      >
        <button
          class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
        >
          -
        </button>
        <h4
          class="text-2xl px-4 py-0"
        >
          0
        </h4>
        <button
          class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
        >
          +
        </button>
      </div>
    </div>
  </div>
</body>
`;
